src/main/java/com/google/devtools/build/lib/packages/BuildType.java

```diff
@@ -558,6 +558,23 @@ public void repr(SkylarkPrinter printer) {
       this(x, what, context, originalType, "");
     }
 
+    /**
+     * This class provides more precise error messaging for type failures on select()s by
+     * highlighting which select() branch has the failure.
+     */
+    private static class CustomToString {
```

Comment:
Is this just to pass to `Object what`? That is, you want something that extends Object and has a toString() that you control?

In that case... how about a String? :D

Reply:
I was racking my brains trying to think of some pre-existing class that did that, and didn't even think of String. Sigh. 

Done!

Reply:
Yeah, that API is a bit confusing, because it looks like it needs some special class or has some special use... but not really, a normal string works just fine. :D;;

Although as I'm saying that, there actually is a benefit - to not have to evaluate `what.toString()` unless the string is needed (because the conversion failed). So if you want to have this class take `key` and `what` and do the formatting in `toString()` in case `what` is some huge object that's really expensive to stringify, that makes sense to me.

Reply:
Point taken. But I vote for simplicity over optimization in this case. Doubly so since this code path only applies to selects.